edu_edfi_airflow/callables/change_version.py

```diff
@@ -0,0 +1,290 @@
+    get_key_changes: bool = False,
+
+    **context
+) -> None:
+    """
+    Overload get_previous_change_versions() with a check against the Ed-Fi API to only return endpoints with new data to process.
+    """
+    return_tuples = get_previous_change_versions(
+        tenant_code=tenant_code, api_year=api_year, endpoints=endpoints,
+        snowflake_conn_id=snowflake_conn_id, change_version_table=change_version_table,
+        get_deletes=get_deletes, get_key_changes=get_key_changes,
+        **context
+    )
+
+    edfi_conn = EdFiHook(edfi_conn_id=edfi_conn_id).get_conn()
+
+    # Only ping the API if the endpoint is specified in the run.
+    config_endpoints = airflow_util.get_config_endpoints(context)
+
+    # Convert (namespace, endpoint) tuples to a {endpoint: namespace} dictionary.
+    endpoint_namespaces = {endpoint: namespace for namespace, endpoint in endpoints}
+
+    # Track which endpoints have deltas or failed total-count gets.
+    logging.info("Checking each endpoint for new records...")
+
+    delta_endpoints = []
+    failed_endpoints = []
+
+    for endpoint, last_max_version in return_tuples:
+
+        # If a subset of endpoints have been selected, only get CV counts for these.
+        if config_endpoints and endpoint not in config_endpoints:
+            continue
+
+        if last_max_version == max_change_version:
+            continue
+
+        namespace = endpoint_namespaces[endpoint]
+
+        try:
+            resource = edfi_conn.resource(
+                endpoint, namespace=namespace,
+                get_deletes=get_deletes, get_key_changes=get_key_changes,
+                min_change_version=last_max_version,
+                max_change_version=max_change_version
+            )
+
+            if not (delta_record_count := resource.total_count()):
+                continue
+
+            logging.info(f"    {namespace}/{endpoint}: {delta_record_count} new records")
+            delta_endpoints.append((endpoint, last_max_version))
+        
+        except Exception:
+            logging.warning(
+                f"    Unable to retrieve record count for endpoint: {namespace}/{endpoint}"
+            )
+            failed_endpoints.append(endpoint)  # Still return the tuples, but mark as failed in the UI.
+            continue
+
+    # Always push the xcom, but return a second xcom if at least one endpoint failed.
+    # This should NOT be necessary, but we encountered a bug where a downstream "none_skipped" task skipped with "upstream_failed" status.
+    if failed_endpoints:
+        logging.info(
+            f"Failed getting delta row count for one or more endpoints: {failed_endpoints}"
+        )
+        context['ti'].xcom_push(key='failed_endpoints', value=failed_endpoints)
+
+    if not delta_endpoints:
+        raise AirflowSkipException("No endpoints to process were found. Skipping downstream ingestion.")
+
+    return delta_endpoints
+
+
+def update_change_versions(
```

Comment:
Setting the `is_key_changes` column is optional to avoid needing to add this column to all implementations immediately, even if they're not using key changes.